### ISSUE TITLE
`toolchain list` shows targets that do not exist

### ISSUE THREAD

**Issue by brson** [rust-lang-nursery/rustup.rs#326]
On stable there aren't yet mips targets but 'toolchain list' still shows them.

At the present this command shows all targets even when the manifest says they aren't available. We might instead hide targets that are indicated as not available. Unfortunately, the manifest is not rich enough to describe *why* the target isn't available, and there are two reasons: either the build failed on a non-gated target, or the target doesn't exist yet on the channel. The 'available' manifest flag was designed for the former use case (transient build failures), not the latter.

Perhaps instead of hiding unavailable toolchains we could instead say `(unavailable)`. Not sure offhand.

**Comment by brson** [comment]
Let's just add "(unavailable)" to the text.

**Comment by Bogay** [comment]
Maybe it has been fixed? At least I can't see mips target in the `rustup +stable target list | grep mips` outputs (version: 1.28.2). Here is the implementation, I can see it filter items by `available` property. 
https://github.com/rust-lang/rustup/blob/28d1352dbcb436d3111c3594b9e1588e94950464/src/cli/rustup_mode.rs#L1280-L1295

**Comment by rami3l** [comment]
@Bogay Yeah, I think we can close this for now thanks to https://github.com/rust-lang/rustup/commit/1a318b122eb1ba0f8fa860e2506acf593b21cdf4 (https://github.com/rust-lang/rustup/pull/4347). Thanks for triaging!
